docs/rules/GH003-no-empty-string-alt.md

```diff
@@ -0,0 +1,35 @@
+# GH003 No empty string alt
+
+## Rule details
+
+⚠️ This rule is _off_ by default and is only applicable for GitHub rendered markdown.
+
+Currently, all images on github.com are automatically wrapped in an anchor tag.
+
+As a result, images that are intentionally marked as decorative (via `alt=""``) end up rendering as a link without an accessible name. This is confusing and inaccessible for assistive technology users. 
+
+This rule can be enabled to enforce that the alt attribute is always set to descriptive text. 
+
+This rule should be removed once this behavior is updated on GitHub's UI.
+
+## Examples
+
+### Incorrect 👎
+
+```md
+![""](cat.png)
```

Comment:
I think a `useful-alt-text` rule might be an interesting avenue to explore, where we ban things like "screenshot", "image", "example", '""', etc. But I think that's a separate rule from empty text.

Reply:
Ah! You're right -- thank you for catching that! I'll stick with HTML for this rule!

Reply:
> useful-alt-text

The [no-default-alt-text](https://github.com/github/markdownlint-github/blob/main/test/no-default-alt-text.test.js) currently flags alternative text like, `image`, `Image`, `Screen Shot 2022-06-26 at 7 41 30 PM` though it's not comprehensive by any means. We could potentially expand on that?

Reply:
Yeah, I think we definitely could expand on that! It might be interesting to see if we can run a query to figure out what the most common alt texts are across GitHub -- I bet the top 20 or so are all things we'd probably want to lint against.